tests/db_functions/test_nullif.py

```diff
@@ -0,0 +1,49 @@
+from unittest import skipIf, skipUnless
+
+from django.db import connection
+from django.db.models import Value
+from django.db.models.functions import NullIf
+from django.test import TestCase
+
+from .models import Author
+
+
+class NullIfTests(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        Author.objects.create(name='John Smith', alias='smithj', sample=bytes(1))
+        Author.objects.create(name='Rhonda', alias='Rhonda', sample=bytes(2))
+
+    @skipIf(connection.vendor == 'oracle', 'Oracle uses the empty string instead')
+    def test_nullif(self):
+        authors = Author.objects.annotate(nullif=NullIf('alias', 'name')).values_list('nullif')
+        self.assertSequenceEqual(authors, [('smithj',), (None,)])
+
+    @skipUnless(connection.vendor == 'oracle', 'Oracle specific test for the NULL return value')
+    def test_nullif_oracle_charfield(self):
+        authors = Author.objects.annotate(nullif=NullIf('alias', 'name')).values_list('nullif')
+        self.assertSequenceEqual(authors, [('smithj',), ('',)])
```

Comment:
I think you can combine `test_nullif` and `test_nullif_oracle_charfield` into single test and use `connection.features.interprets_empty_strings_as_nulls` to determine `empty_value` (see [example](https://github.com/django/django/blob/master/tests/model_forms/tests.py#L278)).